src/nupic/regions/BacktrackingTMRegion.cpp

```diff
@@ -171,7 +169,10 @@ void BacktrackingTMRegion::compute() {
   }
   if (args_.anomalyMode) {
     Array p(NTA_BasicType_Real32, tm_->topDownCompute(), args_.numberOfCols);
-    prevPredictedColumns_ = VectorHelpers::binaryToSparse(p.asVector<Real32>());
+    const vector<Byte> tmpDense = p.asVector<Byte>();
```

Comment:
> Actually, I think what you are trying to do is convert BacktrackingTMRegion to use SDRs. 

not at all. I'm just replacing the VectorHelpers::binaryToSparse with the SDR alternative way to do that. 
I'm not trying for BackTM to just SDR as a whole

Reply:
I guess what I am objecting to is that using the SDR in this way requires at least two passes of the array rather than one and it is not as obvious as to what the code is trying to do.   Frankly I liked the VectorHelpers routines.

But if we assume that eventually BacktrackingTM will be converted to use SDR's correctly then I guess this is ok.

Reply:
> Frankly I liked the VectorHelpers routines. 

ok, maybe I'm forcing this? Should we leave vectorHelpers be? My aim was to eventually replace most/all of that functionality with SDR>